includes/admin/send/class-wpsms-send.php

```diff
@@ -26,39 +28,72 @@ public function __construct()
         $this->options   = Option::getOptions();
     }
 
+
+    public function ajax_get_recipient_counts()
+    {
+        check_ajax_referer('wp_rest', 'wpsms_nonce');
+        $type  = isset($_POST['type']) ? sanitize_text_field($_POST['type']) : '';
+        $value = isset($_POST['value']) ? sanitize_text_field($_POST['value']) : '';
+        $count = 0;
+
+        switch ($type) {
+            case 'roles':
+                $result = Helper::getUsersMobileNumberCountsWithRoleDetails();
+                $count  = $result['roles'][$value]['count'] ?? 0;
+                break;
+
+            case 'wc-customers':
+                $numbers = Helper::getWooCommerceCustomersNumbers();
```

Comment:
We also need to refactor method `getWooCommerceCustomersNumbers` since that queries for 1000 users.